packages/calypso-e2e/src/lib/pages/login-page.ts

```diff
@@ -76,6 +76,22 @@ export class LoginPage {
 		await this.page.goto( getCalypsoURL( 'log-in' ), { waitUntil: 'networkidle' } );
 	}
 
+	/**
+	 * Navigates to the /log-in/{locale} endpoint.
+	 */
+	async visitBlueLogin( locale = '' ): Promise< void > {
+		const targetUrl = locale ? `log-in/${ locale }` : 'log-in';
+		await this.page.goto( getCalypsoURL( targetUrl ), { waitUntil: 'networkidle' } );
+	}
+
+	/**
+	 * Navigates to the /log-in/{locale} endpoint.
+	 */
+	async visitWhiteLogin( locale = '' ): Promise< void > {
+		const targetUrl = locale ? `log-in/new/${ locale }` : 'log-in/new';
+		await this.page.goto( getCalypsoURL( targetUrl ), { waitUntil: 'networkidle' } );
+	}
```

Comment:
It was initially not clear to me (from the JSDoc or method name) what this intended to do.

Now that I understand the intent is to navigate to one of the two variations for the login endpoint, I would like to suggest the following changes to be in line with POM design principles used elsewhere in `calypso-e2e`:

1. an optional parameter be added to the `options?: LoginOptions` interface used by `LoginPage.login`:

```
	async login(
		credentials: LoginCredentials | TestAccount,
		options?: LoginOptions
	): Promise< void > {
```

```
interface LoginOptions {
	landingUrl?: string;
    endpoint?: string;
}
```

2. specify the endpoint when calling `LoginPage.login` like:
```
await loginPage.login( { account: user }, { endpoint: 'log-in/new', landingUrl: '**/log-in/authenticator' } );
```

This way, we avoid creating single-purpose methods in POMs that can easily be integrated into the mainline methods. This helps us avoid the situation we faced with the Selenium-based framework.

Reply:
As suggested, I've refactored the code and reused the existing `visitIO` function to append a path param such that the script navigates to the white background login page(`/log-in/new`) and blue background signup page(`/start/premium`)  based on the props provided to it.